src/lazy/text/buffer.rs

```diff
@@ -931,18 +948,22 @@ impl<'data> TextBufferView<'data> {
     pub fn match_list_value_1_1(
         self,
     ) -> IonParseResult<'data, Option<LazyRawValueExpr<'data, TextEncoding_1_1>>> {
-        alt((
-            whitespace_and_then(
-                terminated(
-                    Self::match_e_expression,
-                    Self::match_delimiter_after_list_value,
-                )
-                .map(|matched| Some(RawValueExpr::MacroInvocation(matched))),
-            ),
-            Self::match_list_value.map(|maybe_matched| {
+        whitespace_and_then(alt((
+            terminated(
+                Self::match_e_expression,
+                Self::match_delimiter_after_list_value,
+            )
+            .map(|matched| Some(RawValueExpr::MacroInvocation(matched))),
+            value(None, tag("]")),
+            terminated(
+                Self::match_annotated_value_1_1.map(Some),
```

Comment:
🗺️ Same problem here as in `match_sexp_value_1_1`. Fixed by copy/pasting the content of `match_list_value`, and altering this line to point to `match_annotated_value_1_1`.